server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/IssueDownloader.java

```diff
@@ -21,108 +21,147 @@
 
 import java.time.Instant;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 import javax.annotation.CheckForNull;
-import javax.annotation.Nullable;
 import org.apache.commons.lang3.StringUtils;
 import org.sonar.scanner.protocol.input.ScannerInput;
+import org.sonarsource.sonarlint.core.commons.Language;
 import org.sonarsource.sonarlint.core.commons.RuleKey;
+import org.sonarsource.sonarlint.core.commons.Version;
 import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
 import org.sonarsource.sonarlint.core.commons.progress.ProgressMonitor;
 import org.sonarsource.sonarlint.core.serverapi.ServerApi;
 import org.sonarsource.sonarlint.core.serverapi.ServerApiHelper;
 import org.sonarsource.sonarlint.core.serverapi.issue.IssueApi;
 import org.sonarsource.sonarlint.core.serverapi.proto.sonarqube.ws.Common.Flow;
 import org.sonarsource.sonarlint.core.serverapi.proto.sonarqube.ws.Common.TextRange;
+import org.sonarsource.sonarlint.core.serverapi.proto.sonarqube.ws.Issues;
 import org.sonarsource.sonarlint.core.serverapi.proto.sonarqube.ws.Issues.Issue;
+import org.sonarsource.sonarlint.core.serverapi.proto.sonarqube.ws.Issues.IssueLite;
+import org.sonarsource.sonarlint.core.serverapi.rules.RulesApi;
 import org.sonarsource.sonarlint.core.serverapi.source.SourceApi;
 import org.sonarsource.sonarlint.core.serverapi.util.ServerApiUtils;
 
 public class IssueDownloader {
 
-  private static final Set<String> NON_CLOSED_STATUSES = new HashSet<>(Arrays.asList("OPEN", "CONFIRMED", "REOPENED"));
-
   private static final SonarLintLogger LOG = SonarLintLogger.get();
 
   /**
    * Fetch all issues of the component with specified key.
    * If the component doesn't exist or it exists but has no issues, an empty iterator is returned.
    *
    * @param key project key, or file key.
-   * @param branchName name of the branch. If null - issues will be downloaded only for the main branch.
+   * @param branchName name of the branch.
    * @return Iterator of issues. It can be empty but never null.
    */
-  public List<ServerIssue> download(ServerApiHelper serverApiHelper, String key, boolean fetchTaintVulnerabilities,
-    @Nullable String branchName,
-    ProgressMonitor progress) {
-    var issueApi = new ServerApi(serverApiHelper).issue();
+  public List<ServerIssue> download(ServerApiHelper serverApiHelper, String key, String branchName, boolean isSonarCloud,
+    Version serverVersion, ProgressMonitor progress) {
```

Comment:
The progress monitor is not used in this method, should we use it or remove it ?